### ISSUE TITLE
[Feature Request] One-shot 100% battery charge

### ISSUE THREAD

**Issue by kosma** [seerge/g-helper#1390]
There's a use case I would love to get proper support for:

- Normally, I charge the battery to 80% to prolong its life.
- Every once in a while I need to travel/see a client/etc. and I want to charged to 100% - but just once.
- But: whenever I set the slider to 100%, I forget to set it back.

It would be nice to have a button that says "Charge the battery to 100%, but only once, afterwards reset to the usual value you set previously". That way I could plug the charger in the day before, press the button, and not have to worry about it anymore.

**Comment by seerge** [comment]
@kosma i can add a button that would set charge limit to 100% w/o remembering it in settings (i.e. app won't reapply it).

As properly waiting for full charge and then turning it off, could be tricky and can also produce some behavior people won't expect or understand.

But I don't have any good spot in whole battery section left for this button (all corners are occupied)

**Comment by kosma** [comment]
I agree that waiting for full charge is difficult, especially with how failing batteries tend to have their SoC all over the place, sometimes never reaching 100%. Maybe instead it could be "set the charge limit to 100% until charger is unplugged"? Then the check is simple and should have less corner cases?

As for the lack of space - I like your idea of "space driven development" - when there's no space left in the GUI, the program is complete :D

**Comment by seerge** [comment]
@kosma it could be a lot of uncontrolled situations, like laptop will get turned off and reach 100% charge in that state (i.e. w/o app running). That's why i intentionally don't want to create any complicated scenarios here

**Comment by seerge** [comment]
@kosma check this build

I have added a new button with tooltip explaining what it does. I'm not a big fan of this UI solution as it makes UI even more cluttered, but I didn't find anything better so far

[GHelper.zip](https://github.com/seerge/g-helper/files/12777009/GHelper.zip)

**Comment by Varriount** [comment]
@seerge Perhaps There should be some event/control API? I'm not super-familiar with C#, but I know that PowerShell has its own event system, and Windows has multiple ways of publishing events. That way additional features like this could be deferred to scripts.

**Comment by seerge** [comment]
@Varriount as I have mentioned above, if laptop would charge to full while being off, not a single API would help as system is powered off ;)

I believe that current implementation (i.e. from build linked here) - when pressing on 100% button would set charge limit too 100% w/o saving it in config is sufficient enough for full charging w/o introducing any not-transparent and difficult to understand logic :)

**Comment by seerge** [comment]
In this build I have turned ``100%`` button into a toggle, so you can turn 100% limit ON and OFF by clicking on it (similar to way it works with ``Fn-Lock`` button in neighboring section)

[GHelper.zip](https://github.com/seerge/g-helper/files/12792054/GHelper.zip)

**Comment by Kowken** [comment]
@seerge I have a question regarding that. So if I understand correctly, this one-time 100% won't take effect if one turns their laptop off right after setting it as it is not saved in the config right ? At least for me it does not (put in on charge at 60% with a 80% power limit and pushed the 100% before turning it off: turned it on at 80% the next morning).
